src/azul/indexer/document.py

```diff
@@ -396,8 +400,19 @@ def translate_fields(cls,
                 return [cls.translate_fields(val, field_types, forward=forward, path=path) for val in doc]
             else:
                 assert False, (path, type(doc))
-        elif isinstance(field_types, FieldType):
-            field_type = field_types
+        else:
+            if isinstance(field_types, list):
+                # FIXME: Assert that a non-list field_type implies a non-list
+                #        doc (only possible for contributions).
+                #        https://github.com/DataBiosphere/azul/issues/2689
+                assert ((forward and doc is None)
```

Comment:
Should probably use `if` expression to avoid the symmetric guards (`forward` and `not forward`).